python/lsst/meas/extensions/psfex/psfexPsfDeterminer.py

```diff
@@ -35,6 +35,54 @@
 import lsst.meas.algorithms as measAlg
 import lsst.meas.algorithms.utils as maUtils
 import lsst.meas.extensions.psfex as psfex
+from lsst.pipe.base import AlgorithmError
+
+
+class PsfexTooFewStarsAlgorithmError(AlgorithmError):
+    """Custom exception for PSFEx algorithmic error when too few good stars are
+    available for PSF determination.
+    """
+
+    def __init__(
+        self,
+        message,
+        num_available_stars,
+        num_good_stars,
+        poly_ndim_initial,
+        poly_ndim_final,
+    ) -> None:
+        """
+        Initialize the PSFEx error exception for too few stars.
+
+        Parameters
+        ----------
+        message : str
+            Error message.
```

Comment:
Instead of passing it in, I'd build the message at init, since this exception is only raised in a specific context. Then you can include other information from the other passed-in parameters in the message.

Reply:
Done!